File: src/main/java/tn/esprit/foyer/Entity/Foyer.java
package tn.esprit.foyer.Entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.*;

import java.io.Serializable;
import java.util.List;


@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString

public class Foyer implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idFoyer;
    private String nomFoyer;
    @Column(columnDefinition = "boolean Default False")
    private boolean archiverFoyer;
    private Long capacityFoyer;
    @OneToOne(cascade = CascadeType.ALL,mappedBy = "foyer")
    @JsonManagedReference

    private Universite universite;
    @OneToMany(cascade = CascadeType.ALL, mappedBy="foyer")
    private List<Bloc> Blocs;
}

File: src/app/models/Foyer.ts
import { Universite } from '../models/Universite';
import { Bloc } from '../models/Bloc';

export class Foyer {
    idFoyer: number;
    nomFoyer: string;
    archiverFoyer: boolean;
    capacityFoyer: number;
    universite?: Universite;
    blocs?: Bloc[];

    constructor(
        idFoyer: number,
        nomFoyer: string,
        archiverFoyer: boolean,
        capacityFoyer: number,
        universite?: Universite,
        blocs?: Bloc[]
    ) {
        this.idFoyer = idFoyer;
        this.nomFoyer = nomFoyer;
        this.archiverFoyer = archiverFoyer;
        this.capacityFoyer = capacityFoyer;
        this.universite = universite;
        this.blocs = blocs;
    }
}
